src/libaktualizr/http/httpclient.cc

```diff
@@ -73,7 +61,7 @@ HttpClient::HttpClient(const std::vector<std::string>* extra_headers)
     }
   }
   curlEasySetoptWrapper(curl, CURLOPT_HTTPHEADER, headers);
-  curlEasySetoptWrapper(curl, CURLOPT_USERAGENT, user_agent.c_str());
+  curlEasySetoptWrapper(curl, CURLOPT_USERAGENT, Utils::getUserAgent());
```

Comment:
It might override the user agent header specified by an HttpClient user/client in `extra_headers`.
IMHO, statically predefined headers should be specified in a HTTP client factory configuration  (which doesn't exist), so libaktualizr clients could specify desired mandatory headers in explicit way. Utils::set/getUserAgent() is an implicit API for libaktualizr clients/apps they need to be aware and set it before Aktualizr instantiation. 
This is just my two cents and can be ignored since it will require more work :)

Reply:
Yes, I agree this is not especially elegant. The main problem I had is that different garage tools are using some common code that instantiates an HttpClient, that's why I opted for the less intrusive global setting.

I don't see much use case for HttpClient users redefining user agents dynamically though, it's maybe the (rare) kind of thing that can be setup globally.

But then I also question my choice of putting the API in utils.h. I'll try moving it to http client itself.

Reply:
No problem, I understand that we need to be practical and balance between an ideal but time consuming and a just enough but quicker solutions.
BTW, at first glance, this PR, IMHO, is related to this one https://github.com/advancedtelematic/aktualizr/issues/1273#issuecomment-520352928

Reply:
lol - i've been watching this PR to see how things went before pushing my change which will at best produce a merge conflict.

Reply:
Haha well... This PR is not really urgent anyway. Would you also prefer that to be done another way?

Reply:
no. I was just wanted to see the approach that was chosen here since there's a good chance I could do the same thing in my change.